File: Python/bubbleSort.py
#Python program to perform bubble sort
def bubbleSort(array):
    n = len(array)

    for i in range(n-1):
        noSwaps = True
        for j in range(0,n-i-1):
            if array[j] > array[j+1]:
                #Swap
                array[j],array[j+1] = array[j+1],array[j]
                noSwaps = False
        if noSwaps:
            break

if __name__ == "__main__":
    print("\n------BUBBLE SORT------\n")
    n = int(input("Enter array size: "))

    array = []

    print("Insert array elements: ")
    for i in range(n):
        ele = int(input())
        array.append(ele)

    bubbleSort(array)

    print("Sorted array: ")
    for i in range(n):
        print(array[i])

File: C/bubbleSort.c
#include<stdio.h>
#include<stdlib.h>

void main(){
    int n;
    printf("\n----------BUBBLE SORT----------\n\n");
    printf("Enter array size: ");
    scanf("%d",&n);
    int *array;
    array = (int*)malloc(sizeof(int)*n);
    printf("Insert array elements: ");

    for(int i=0;i<n;i++){
        scanf("%d",&array[i]);
    }

    int i,j,temp;
    //bubble sort
    for(i=0;i<n-1;i++){
        int noSwaps = 1;
        for(j=0;j<n-1-i;j++){
            if(array[j] > array[j+1]){
                //swap
                temp = array[j+1] + array[j];
                array[j+1] = temp - array[j+1];
                array[j] = temp - array[j];
                noSwaps = 0;
            }
        }
        if (noSwaps)
            break;
    }

    printf("Sorted array: ");
    for(int i=0;i<n;i++){
        printf("%d ",array[i]);
    }
    printf("\n");
    printf("\n--------------EXIT--------------\n");
}